convert2rhel/subscription.py

```diff
@@ -281,18 +281,30 @@ def attach_subscription():
         tool_opts.pool = pool
         loggerinst.info("Attaching provided subscription pool ID to the system ...")
     else:
-        # Let the user choose the subscription appropriate for the conversion
-        loggerinst.info("Manually select subscription appropriate for the conversion")
         subs_list = get_avail_subs()
+
         if len(subs_list) == 0:
             loggerinst.warning("No subscription available for the conversion.")
             return False
 
-        print_avail_subs(subs_list)
-        sub_num = utils.let_user_choose_item(len(subs_list), "subscription")
-        pool = "--pool " + subs_list[sub_num].pool_id
-        tool_opts.pool = pool
-        loggerinst.info("Attaching subscription with pool ID %s to the system ..." % subs_list[sub_num].pool_id)
+        if len(subs_list) == 1:
+            pool = "--pool " + subs_list[0].pool_id
+            tool_opts.pool = pool
+
+            loggerinst.info(
+                "This %s was the only subscription that you have, It will automaticlly be attached for the conversion."
+                % subs_list[0].pool_id
+            )
+
+        else:
+            # Let the user choose the subscription appropriate for the conversion
+            loggerinst.info("Manually select subscription appropriate for the conversion")
+
+            print_avail_subs(subs_list)
+            sub_num = utils.let_user_choose_item(len(subs_list), "subscription")
+            pool = "--pool " + subs_list[sub_num].pool_id
+            tool_opts.pool = pool
+            loggerinst.info("Attaching subscription with pool ID %s to the system ..." % subs_list[sub_num].pool_id)
```

Comment:
Just a suggestion for a small cleanup:  the code for setting the pool is almost a duplicate in both branches of the if statement.  it should be possible to move that outside the if statement so you only have to maintain one copy of it like this:

``` python 
        if len(subs_list) == 1:
            sub_num = 1
            # Remove pool = "--pool" [..]
             # and tool_opts.pool = [..]
            loggerinst.info([...])
        else:
            [..]
            # Remove pool = "--pool" [..]
             # and tool_opts.pool = [..]

     # Setting pool and tool_opts can now be done here, outside of the if-statement:
     pool = "--pool " + subs_list[sub_num].pool_id
     tool_opts.pool = pool
     _, ret_code = utils.run_subprocess("subscription-manager attach %s" % pool)
```

Reply:
gotcha, I will add these changes and re-push!